### ISSUE TITLE
Game becomes seconds per frame when nullifer encounters a buffed teleporter zone with Aetherium.

### ISSUE THREAD

**Issue by Darkmega18** [JoshTClark/RoR2Mod#7]
Aetherium creates shrines that give you buff fires that orbit you. When you give them to the teleporter when it starts up, the teleporter radius gives the buffs to allies or debuffs to enemies making it easier to hold your ground.

However.

When nullifiers exist anywhere within the area because the debuffs are being refreshed every possible tick probably, and your nullifiers aren't giving a damn due to their balancing issues of being entirely immune and constantly self purging you end up with an overloaded game that shits itself and immediately becomes a slide show. :<

**Comment by KaitoXion** [comment]
It may or may not.
For my case, I did not use Aetherium.
However, after massive amount of enemy spawned, even after I killed most of them; the game lag still keep getting worse.

**Note to DEV**
Meaning is highly very likely that the causes is due to memory of the certain dead enemies is still marked as alive aside from the animation showing that it is dead.
Is also possible that the debuff stat of the enemy still remains even after death causing duplicates and keep increasing non stop.

Unless is speed running, this can be detected otherwise.

Anyway, am still testing. Takes a while as it requires build up of many enemies.

**Comment by KaitoXion** [comment]
> **Note to DEV**
> Meaning is highly very likely that the causes is due to memory of the certain dead enemies is still marked as alive aside from the animation showing that it is dead.
> Is also possible that the debuff stat of the enemy still remains even after death causing duplicates and keep increasing non stop.

Yes. I notice my memory usage is unusually high. Well, I think I'll remove it until its decided to be official.
how did steam use that much memory at 200MB when doing nothing with it?

Confirmed. The memory usage after the game is unusually high. Even after shutting down the ror2 game, it is still using the memory, had to restart.
I suggest to test and observe the memory usage. task manager didn't seems to show a clear reasons for it aside from assigning random large size memory in numbers to multiple application for no reason as those application e.g. chat app do not use that much memory before this.

Kindly let me know if I am wrong and if anyone were able to use the mod without the same problem.
The easiers way to test in my opinion is to use, ### **MOD:** _SmarterEnemies, TweakAI, EnemyAugments, AugmentedBosses and (optional I think) RPGMod_

 Cause you know, it's MS OS these days. Am using win 11. Things from MS is going for the worse each and everything they releases.

**Comment by KomradeSpectre** [comment]
After review, consider hooking into Character Body. Addbuff and Character Body. AddTimedBuff. This will result in less get component calls and more performant code related to debuff negation

You should be able to straight out negate it before calling orig
